cibuildwheel/options.py

```diff
@@ -345,12 +351,11 @@ def config_file_path(self) -> Optional[Path]:
 
         return None
 
-    @property
+    @property  # type: ignore[no-redef, misc]
+    @functools.lru_cache(maxsize=None)
     def package_requires_python_str(self) -> Optional[str]:
-        if not hasattr(self, "_package_requires_python_str"):
-            args = self.command_line_arguments
-            self._package_requires_python_str = get_requires_python_str(Path(args.package_dir))
-        return self._package_requires_python_str
+        args = self.command_line_arguments
+        return get_requires_python_str(Path(args.package_dir))
```

Comment:
I'm kinda surprised we haven't got a `@cached_property` decorator in this project. I wonder if mypy would like that better...?

Reply:
`cached_property` is 3.8+ only.

Reply:
I need to think about this and make sure I didn't leak here. Not that it would really hurt us, but would be a bad example.

Reply:
we can keep this, though, if you prefer.

Reply:
I'm still working though the code, but I think I might have created a leak here with the current code - https://youtu.be/sVjtp6tGo0g?list=PLWBKAf81pmOaP9naRiNAqug6EBnkPakvY. It's probably better to vendor in a cached_property if you are fine with that.

Reply:
Vendored a copy from the stdlib + typeshed.